src/components/views/rooms/wysiwyg_composer/hooks/useSuggestion.ts

```diff
@@ -59,38 +58,128 @@ export function useSuggestion(
 } {
     const [suggestion, setSuggestion] = useState<SuggestionState>(null);
 
-    // TODO handle the mentions (@user, #room etc)
-    const handleMention = (): void => {};
-
-    // We create a `seletionchange` handler here because we need to know when the user has moved the cursor,
+    // We create a `selectionchange` handler here because we need to know when the user has moved the cursor,
     // we can not depend on input events only
-    const onSelect = (): void => processSelectionChange(editorRef, suggestion, setSuggestion);
+    const onSelect = (): void => processSelectionChange(editorRef, setSuggestion);
+
+    const handleMention = (href: string, displayName: string, attributes: Attributes): void =>
+        processMention(href, displayName, attributes, suggestion, setSuggestion, setText);
 
     const handleCommand = (replacementText: string): void =>
         processCommand(replacementText, suggestion, setSuggestion, setText);
 
+    const memoizedMappedSuggestion: MappedSuggestion | null = useMemo(() => {
+        return suggestion !== null
+            ? { keyChar: suggestion.keyChar, type: suggestion.type, text: suggestion.text }
+            : null;
+    }, [suggestion]);
+
     return {
-        suggestion: mapSuggestion(suggestion),
+        suggestion: memoizedMappedSuggestion,
         handleCommand,
         handleMention,
         onSelect,
     };
 }
 
 /**
- * Convert a PlainTextSuggestionPattern (or null) to a MappedSuggestion (or null)
+ * When the selection changes inside the current editor, check to see if the cursor is inside
+ * something that could be a command or a mention and update the suggestion state if so
  *
- * @param suggestion - the suggestion that is the JS equivalent of the rust model's representation
- * @returns - null if the input is null, a MappedSuggestion if the input is non-null
+ * @param editorRef - ref to the composer
+ * @param setSuggestion - the setter for the suggestion state
  */
-export const mapSuggestion = (suggestion: SuggestionState): MappedSuggestion | null => {
```

Comment:
This function was used once, so effectively implemented it as an anonymous function inline in the call to `useMemo` inside `useSuggestion`